include/gridtools/stencil_composition/backend_cuda/tmp_storage_sid.hpp

```diff
@@ -12,128 +12,56 @@
+            template <class NumColors, class IBlockSize, class JBlockSize, class Extent>
+            auto sizes(NumColors num_colors,
+                IBlockSize i_block_size,
+                JBlockSize j_block_size,
+                Extent,
+                int_t n_blocks_i,
+                int_t n_blocks_j,
+                int_t k_size) {
+                return tuple_util::make<
+                    hymap::keys<dim::i, dim::j, dim::c, sid::blocked_dim<dim::i>, sid::blocked_dim<dim::j>, dim::k>::
+                        values>(Extent::extend(dim::i(), i_block_size),
+                    Extent::extend(dim::j(), j_block_size),
+                    num_colors,
+                    n_blocks_i,
+                    n_blocks_j,
+                    Extent::extend(dim::k(), k_size));
+            }
+        } // namespace tmp_impl_
 
-    namespace tmp_cuda_impl_ {
-        template <class Extent>
-        hymap::keys<dim::i, dim::j>::values<integral_constant<int_t, -Extent::iminus::value>,
-            integral_constant<int_t, -Extent::jminus::value>>
-        origin_offset(Extent) {
-            return {};
-        }
-    } // namespace tmp_cuda_impl_
-
-#ifndef GT_ICOSAHEDRAL_GRIDS
-
-    namespace tmp_cuda_impl_ {
-        template <int_t ComputeBlockSizeI,
-            int_t ComputeBlockSizeJ,
-            int_t ExtentIMinus, // negative by convention
-            int_t ExtentIPlus,
-            int_t ExtentJMinus, // negative by convention
-            int_t ExtentJPlus,
-            int_t ExtentKMinus, // negative by convention
-            int_t ExtentKPlus,
-            class BlockSizeI = integral_constant<int_t, ComputeBlockSizeI - ExtentIMinus + ExtentIPlus>,
-            class BlockSizeJ = integral_constant<int_t, ComputeBlockSizeJ - ExtentJMinus + ExtentJPlus>>
-        auto sizes(tmp_cuda::blocksize<ComputeBlockSizeI, ComputeBlockSizeJ>,
-            extent<ExtentIMinus, ExtentIPlus, ExtentJMinus, ExtentJPlus, ExtentKMinus, ExtentKPlus>,
+        template <class Data, class NumColors, class BlockSizeI, class BlockSizeJ, class Extent, class Allocator>
+        auto make_tmp_storage(NumColors num_colors,
+            BlockSizeI block_size_i,
+            BlockSizeJ block_size_j,
+            Extent extent,
             int_t n_blocks_i,
             int_t n_blocks_j,
-            int_t k_size) {
-            return tuple_util::make<
-                hymap::keys<dim::i, dim::j, sid::blocked_dim<dim::i>, sid::blocked_dim<dim::j>, dim::k>::values>(
-                BlockSizeI{}, BlockSizeJ{}, n_blocks_i, n_blocks_j, k_size);
+            int_t k_size,
+            Allocator &alloc) {
+            GT_STATIC_ASSERT(is_extent<Extent>::value, GT_INTERNAL_ERROR);
+            return sid::shift_sid_origin(
+                sid::make_contiguous<Data, int_t>(alloc,
+                    tmp_impl_::sizes(num_colors, block_size_i, block_size_j, extent, n_blocks_i, n_blocks_j, k_size)),
+                tmp_impl_::origin_offset(extent));
```

Comment:
There doesn’t seem to be any alignment/padding implemented, right? _Maybe_ this could improve performance. On vertical advection in the standalone benchmarks, disabling alignment (for all storages, not only temporaries), reduces performance for almost 40%. As most storages there are no temporaries but constant fields (which are aligned anyway in GridTools), the impact might be much much smaller, but still: alignment/padding can improve performance significantly.

Reply:
I completely agree but  for now I've only added TODO.